tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -32,18 +32,14 @@ void EnsureUniqueNTupleNames(const std::vector<RNTupleOpenSpec> &ntuples)
 } // anonymous namespace
 
 std::unique_ptr<ROOT::Experimental::RNTupleProcessor>
-ROOT::Experimental::RNTupleProcessor::Create(const RNTupleOpenSpec &ntuple)
+ROOT::Experimental::RNTupleProcessor::Create(const RNTupleOpenSpec &ntuple, std::unique_ptr<RNTupleModel> model)
 {
-   auto pageSource = Internal::RPageSource::Create(ntuple.fNTupleName, ntuple.fStorage);
-   pageSource->Attach();
-   auto model = pageSource->GetSharedDescriptorGuard()->CreateModel();
-   return RNTupleProcessor::Create(ntuple, *model);
-}
-
-std::unique_ptr<ROOT::Experimental::RNTupleProcessor>
-ROOT::Experimental::RNTupleProcessor::Create(const RNTupleOpenSpec &ntuple, RNTupleModel &model)
-{
-   return std::unique_ptr<RNTupleSingleProcessor>(new RNTupleSingleProcessor(ntuple, model));
+   if (!model) {
+      auto pageSource = Internal::RPageSource::Create(ntuple.fNTupleName, ntuple.fStorage);
```

Comment:
This means creating the page source twice, can this be delayed until the processor is connected (as the chain and join processors do)?